examples/tracingpolicy/process-exec/process-exec-elf-begin.yaml

```diff
@@ -0,0 +1,27 @@
+# This tracing policy 'process-exec-elf-begin' report the elf binary path
+# being executed, just when we are switching execution to it.
+#
+# Description:
+#  Report the file path of the new elf or flat binary that is being executed.
+#
+# Limitation:
+#  This does not report different binary format handlers or scripts paths that
+#  are being executed, but will report the final elf or flat binary, like
+#  shebang (!#/bin/bash) used to handle the script.
+#
+apiVersion: cilium.io/v1alpha1
+kind: TracingPolicy
+metadata:
+  name: "process-exec-elf-begin"
+  #annotations:
+    #description: "Report the path of the elf or flat binary that is being executed"
+    #author: "Djalal Harouni"
+spec:
+  kprobes:
+  - call: "security_bprm_creds_from_file"
```

Comment:
question for me that might be interesting for other people when looking at this: what is this hook? I found this piece of doc https://www.kernel.org/doc/html/latest/core-api/kernel-api.html?highlight=memcpy#c.security_bprm_creds_from_file. When does that happen in the execution chain? And why use this particular symbol?

Reply:
this happens when executing the real binary or elf binary, to compute which UIDs to use for the exection, so like stat binary take those uids, and use it to construct the creds => credentials.